contracts/Ovm_SpokePool.sol

```diff
@@ -141,17 +143,23 @@ contract Ovm_SpokePool is CrossDomainEnabled, SpokePool {
             WETH9(relayerRefundLeaf.l2TokenAddress).withdraw(relayerRefundLeaf.amountToReturn); // Unwrap into ETH.
             relayerRefundLeaf.l2TokenAddress = l2Eth; // Set the l2TokenAddress to ETH.
         }
-        IL2ERC20Bridge(
-            tokenBridges[relayerRefundLeaf.l2TokenAddress] == address(0)
-                ? Lib_PredeployAddresses.L2_STANDARD_BRIDGE
-                : tokenBridges[relayerRefundLeaf.l2TokenAddress]
-        ).withdrawTo(
-                relayerRefundLeaf.l2TokenAddress, // _l2Token. Address of the L2 token to bridge over.
-                hubPool, // _to. Withdraw, over the bridge, to the l1 pool contract.
-                relayerRefundLeaf.amountToReturn, // _amount.
-                l1Gas, // _l1Gas. Unused, but included for potential forward compatibility considerations
-                "" // _data. We don't need to send any data for the bridging action.
-            );
+
+        address l2TokenAddress = relayerRefundLeaf.l2TokenAddress;
+        address tokenBridge = tokenBridges[l2TokenAddress] == address(0)
+            ? Lib_PredeployAddresses.L2_STANDARD_BRIDGE
+            : tokenBridges[l2TokenAddress];
+        // In some cases such as SNX, the custom token bridge might require approval to transfer the ERC-20 before
+        // it can be burnt. We can grant approval and immediately revoke after the call to ensure no lingering
+        // approval remains.
+        IERC20(l2TokenAddress).approve(tokenBridge, relayerRefundLeaf.amountToReturn);
```

Comment:
Downside to this approach is extra gas per bridge call, but not sure there is a different way if we go through with this approach

Reply:
Agree but the gas increase should be minimal though, and also much cheaper considering this is Optimism as well.